### PR TITLE
Add insn cmd to interactive debug mode

### PR DESCRIPTION
**Pull request by abejgonzalez**
This PR adds a new interactive debug mode cmd of `insn` with the same API as the `pc` command to print out the instruction associated with the PC of a specific hart. This command can be used with the `until*` commands to run a binary "until" the "insn" value is a specific value.

### CODE REVIEW

**Review by aswaterman — COMMENTED** [review]
This looks like a reasonable addition to me, but do you think it makes sense to enhance it by printing out the disassembled instruction after printing out its opcode?  I guess it doesn't matter much since the main purpose is the `until` aspect, not the printing aspect, so I'll leave it up to you.

**Comment by abejgonzalez** [comment]
> This looks like a reasonable addition to me, but do you think it makes sense to enhance it by printing out the disassembled instruction after printing out its opcode? I guess it doesn't matter much since the main purpose is the `until` aspect, not the printing aspect, so I'll leave it up to you.

It would be nice. I have a commit ready to add it but I'm unsure of the format for output. Right now it looks like the following:

```bash
(spike) insn 0
0x0202859300000297 auipc   t0, 0x0
```

If there is no preference on the format (i.e. `<HEX> <DISASM>`) then I can commit it.

**Comment by abejgonzalez** [comment]
> > This looks like a reasonable addition to me, but do you think it makes sense to enhance it by printing out the disassembled instruction after printing out its opcode? I guess it doesn't matter much since the main purpose is the `until` aspect, not the printing aspect, so I'll leave it up to you.
> 
> It would be nice. I have a commit ready to add it but I'm unsure of the format for output. Right now it looks like the following:
> 
> ```shell
> (spike) insn 0
> 0x0202859300000297 auipc   t0, 0x0
> ```
> 
> If there is no preference on the format (i.e. `<HEX> <DISASM>`) then I can commit it.

Done.

File: riscv/interactive.cc
Comment (abejgonzalez) [inline riscv/interactive.cc]: Currently this is a bit duplicated with `get_pc` and `get_mem`, but this is significantly faster since reusing those functions requires you to re-create a `vector<string>`'s to use them.

Comment (aswaterman) [reply riscv/interactive.cc]: I think it's OK... no more sinful than everything else in `interactive.cc`.

**Comment by aswaterman** [comment]
Yeah, that's fine.

Can you squash these commits into one?

**Comment by aswaterman** [comment]
Also note that CI is broken until #1710 is merged.

**Comment by abejgonzalez** [comment]
> Yeah, that's fine.
> 
> Can you squash these commits into one?

Will do.

File: riscv/interactive.cc
Comment (aswaterman) [reply riscv/interactive.cc]: Actually, can you do some light refactoring?  Hoist this out as its own function `static reg_t load(mmu_t* mmu, reg_t addr)` so you can share the code between the two places without perf loss.

Comment (abejgonzalez) [reply riscv/interactive.cc]: Done.

**Review by aswaterman — APPROVED** [review]
LGTM.

**Pull request merged**
